### ISSUE TITLE
import/export - Multiple exports triggered when duplication isn't relevant

### ISSUE THREAD

**Issue by tomprats** [benmosher/eslint-plugin-import#1704]
I get the error `Multiple exports of name 'List'` from both lines of `components/pages/index.js` in following example:

```
// components/pages/index.js
export * as Messages from "./messages";
export * as Users from "./users";

// components/pages/messages/index.js
export List from "./list";
export Show from "./show";

// components/pages/users/index.js
export Edit from "./edit";
export List from "./list";

// components/pages/messages/list.js => React Component
// components/pages/messages/show.js => React Component
// components/pages/users/edit.js => React Component
// components/pages/users/list.js => React Component
```

I understand that if I get it once, I'll get it twice, but I don't think it should be happening at all. The code itself works when I access my components (`Pages.Users.List`) leading me to believe this is a bug in the rule.

**Comment by ljharb** [comment]
When you say "the code works", do you mean, in native browser or node ESM, or in babel/typescript?

**Comment by ljharb** [comment]
In other words, I believe the spec requires this to be a syntax error, because you have two exports that conflict. Babel might paper over this by silently choosing the last one, but that doesn't mean it's valid.

**Comment by tomprats** [comment]
Oh yeah, my bad. I'm using babel and I think the specific relevant plugin is `plugin-proposal-export-namespace-from` which looks like it's stage 4. 

I think that might change the context of how it works, but do they conflict? With that plugin I can access both `Pages.Messages.List` and `Pages.Users.List`. I thought it was fine because it'd immediately assign the `*` export to the namespace (`Messages` or `Users`) so their contents wouldn't pollute anything

**Comment by tomprats** [comment]
Also in case it's not clear, the commented out portions above are meant to denote individual files. They're not all in one file

**Comment by golopot** [comment]
> When you say "the code works", do you mean, in native browser or node ESM, or in babel/typescript?

The example works without syntax error in node ESM. The `index.js` module has two exports, `Messages` and `Users`. But our plugin reports `Multiple exports of name 'List'`. Something is wrong here.

**Comment by CatsMiaow** [comment]
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#export-star-as-namespace-syntax
The `export * as` was added in TypeScript v3.8

[Playground](https://www.typescriptlang.org/play/?module=1#code/KYDwDg9gTgLgBAKjgQwM5wLLFa5BzbOAMyggFs4AiAOgHoztcDVKBuAKFElkRXQFVUwKOhLkqdAK5CRbdkA)
```ts
// ts
export * as Messages from "./messages";
export * as Users from "./users";
// js
exports.Messages = require("./messages");
exports.Users = require("./users");
```
Using `export * as` does not conflict even if the property names are the same.
No `Multiple exports of name 'List'` errors should occur.

**Comment by ljharb** [comment]
Ah, very true - `export * as X` can't conflict with anything but another `X`.

This definitely seems like a bug.

**Comment by thecotne** [comment]
@ljharb can you remove typescript label since `Proposal: export ns from` is part of standard js https://github.com/tc39/ecma262/pull/1174

**Comment by danrivett** [comment]
Also ran into this issue with the following code:

```ts
export * as WorkerPhoneCreated from './phone/created';
export * as WorkerPhoneUpdated from './phone/updated';
```

Both the following imported files contain an `EventType` export, and should be referenced as `WorkerPhoneCreated.EventType` and `WorkerPhoneUpdated.EventType` respectively and so I think this "multiple exports" error is a false positive if I've understood correctly.

**Comment by ljharb** [comment]
I agree; that should be addressed.

**Comment by lifeiscontent** [comment]
@ljharb any updates on this? I just ran into this recently.

**Comment by ljharb** [comment]
@lifeiscontent no; nobody's sent a PR yet and I haven't had the time. One would be welcome :-)

**Comment by ftaiolivista** [comment]
Still getting this error on v 2.23.4

```
// D.js 
const f = () => 'D::f'
export { f }

// E.js
export * from './D.js'
const f = () => 'E::f'
export { f }
```

-> import/export: Multiple exports of name 'f'.

**Comment by thecotne** [comment]
@ftaiolivista that is legit error.

**Comment by ftaiolivista** [comment]
I think the export overloading is ammisible by ECMASCRIPT (see this answer https://stackoverflow.com/questions/66533367/import-export-name-collision-resolution )

Maybe eslint could give an optionable error that can ben disabled by config?

**Comment by ljharb** [comment]
@ftaiolivista i agree; could you file a new issue for that?
